### PR TITLE
Implementation

### PR DESCRIPTION
**Pull request by CherednychenkoViktoriia**

### CODE REVIEW

File: WinException.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+#include <iostream>
+#include <Windows.h>
+
+class WinException : public std::runtime_error {
+public:
+    WinException(const char* message, DWORD errorMessageID) :
```
Comment (ak-toktor) [inline WinException.h:7]: U can create constructor with with one parameter message and call GetLastError from the constructor

File: WinException.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+#include <iostream>
```
Comment (ak-toktor) [inline WinException.h:2]: unused header

File: Client_Server.cpp [hunk 3]
```diff
@@ -1,6 +1,168 @@
...
+    }
+    ~Server() {
+        m_exit = true;
+        BOOL resWaitForThreadServer = WaitForSingleObject(m_threadServer, INFINITE);
+        if (resWaitForThreadServer == WAIT_FAILED) {
+            std::string str = GetLastErrorAsString();
+            std::cout << str << std::endl;
+        }
+        BOOL resCloseClientEventHandle = CloseHandle(m_clientEvent);
+        if (resCloseClientEventHandle == 0) {
```
Comment (ak-toktor) [inline Client_Server.cpp:95]: it is better to use:
if(!CloseHandle(m_clientEvent) || !CloseHandle(m_resultsEvent) || !CloseHandle(m_threadServer)) {
    std::cout << GetLastErrorAsString() << std::endl;
}

**Review by ak-toktor — CHANGES_REQUESTED** [review]

File: Client_Server.cpp (hunk 3 above)
Comment (CherednychenkoViktoriia) [reply Client_Server.cpp]: This change in code creates an error: Exception thrown at 0x511CFEBA in Client_Server.exe: 0xC0000008: An invalid handle was specified.

**Pull request merged**